qcodes/instrument_drivers/tektronix/keithley_7510.py

```diff
@@ -1,10 +1,372 @@
+
+    @property
+    def data(self) -> DataArray7510:
+        return self._get_data()
+
+    def get_last_reading(self) -> str:
+        """
+        This method requests the latest reading from a reading buffer.
+
+        """
+        if not self.elements():
+            return self.ask(f":FETCh? '{self.short_name}'")
+        fetch_elements = [
+            self.buffer_elements[element] for element in self.elements()
+        ]
+        return self.ask(
+            f":FETCh? '{self.short_name}', {','.join(fetch_elements)}"
+        )
+
+    def _get_data(self) -> DataArray7510:
+        """
+        This command returns the data in the buffer, depends on the user
+        selected elements.
+        """
+        try:
+            _ = self.setpoints()
+        except NotImplementedError:
+            # if the "setpionts" has not been implemented, use a time series
+            # with parameters "t_start" and "t_stop":
+            self.set_setpoints(self.t_start, self.t_stop)
+
+        if self.parent.digi_sense_function() == "None":
+            # when current sense is not digitize sense
+            sense_function = self.parent.sense_function()
+            unit = Sense7510.function_modes[sense_function]["unit"]
+        else:
+            # when current sense is digitize sense
+            sense_function = self.parent.digi_sense_function()
+            unit = DigitizeSense7510.function_modes[sense_function]["unit"]
+
+        elements_units = {
+            "date": "str",
+            "measurement_formatted": "str",
+            "fractional_seconds": "s",
+            "measurement": unit,
+            "relative_time": "s",
+            "seconds": "s",
+            "measurement_status": "",
+            "time": "str",
+            "timestamp": "str",
+            "measurement_unit": "str"
+        }
+
+        if not self.elements():
+            raw_data = self.ask(f":TRACe:DATA? "
+                                f"{self.data_start()}, "
+                                f"{self.data_end()}, "
+                                f"'{self.short_name}'")
+        else:
+            elements = \
+                [self.buffer_elements[element] for element in self.elements()]
+            raw_data = self.ask(f":TRACe:DATA? {self.data_start()}, "
+                                f"{self.data_end()}, "
+                                f"'{self.short_name}', "
+                                f"{','.join(elements)}")
+
+        all_data = raw_data.split(",")
+
+        if len(self.elements()) == 0:
+            elements = ['measurement']
+        else:
+            elements = self.elements()
+        n_elements = len(elements)
+
+        units = tuple(elements_units[element] for element in elements)
+        processed_data = dict.fromkeys(elements)
+        for i, (element, unit) in enumerate(zip(elements, units)):
+            if unit == 'str':
+                processed_data[element] = all_data[i::n_elements]
```

Comment:
This then needs to be converted to the right type? As in to numpy array of floats ?

Reply:
Should I try to converted the non-numerical data to floats, like the datetime "10/20/2020 06:11:11.977197606"?

Reply:
Update: all the non-numerical data is left as string.